include/seqan/stream/iostream_bgzf.h

```diff
@@ -444,6 +448,14 @@ class basic_unbgzf_streambuf :
                             (char*)&job.inputBuffer[0] + BGZF_BLOCK_HEADER_LENGTH,
                             tailLen);
 
+                        // Check if end-of-file marker is set
+                        if (memcmp(reinterpret_cast<uint8_t const *>(&job.inputBuffer[0]),
+                                   reinterpret_cast<uint8_t const *>(&BGZF_END_OF_FILE_MARKER[0]),
+                                   28) == 0)
```

Comment:
Do we now check *every block* for the EOF marker?

Reply:
This is a safety guard.
The specification says, that applications that execute in append mode, can adde EOF marker, but then add new blocks to it. So in that particular case it would be insufficient to stop just here. In addition the followed uncompressed block must have zero length to indicate the true end of the file. Detecting the EOF marker followed by an empty block would be safe indicator for the end of the file. So we could avoid the check here, but I am not sure if this will have a measurable effect on  Bam-IO. I will try a microbenchmark to see if it has any effect.